specs/docs/security/DataHubSecurity.md

```diff
@@ -0,0 +1,266 @@
+  //...
+  }
+);
+```
+### When To Grant Privileges To User Roles And When To Use Amps
+
+User roles should be assigned privileges when the privilege is sufficiently restrictive and does not allow arbitrary code execution.
+
+If the privilege would allow arbitrary code execution, that is a sure indication that an amp should be used. For other privileges, consider if the privilege can be used beyond the intended purpose. Where possible use granular privileges (See [Granular Privileges](https://docs.marklogic.com/guide/security/granular)). If the user role can't use granular privileges and the required privilege would be too broad, (1) a new narrow execute privilege should be created, (2) the function that requires the privilege should check for the narrow privilege, and (3) the function should be amped with the broader privilege.
+
+Roles to be used with amps should be suffixed with `-internal` to indicate that the role is used for amps and NOT intended to be assigned to users.
+
+## Data Hub Roles and Privileges
+
+### data-hub-common
+
+Data Hub role containing the common privileges across Data Hub roles.
+
+#### Inherited Roles
+ - data-hub-module-reader
+ - rest-reader
+
+#### Privileges
+
+Privilege | Notes
+---------|----------
+ xdmp-eval-in | Required to specify database for REST Client. (Note: This privilege alone doesn't allow eval to be called. The REST API code does the neccesary amping for eval execution)
+ xdbc-eval-in | Same as xdmp-eval-in
+
+### data-hub-common-writer
+
+Data Hub role containing the common privileges across Data Hub writer roles.
+
+#### Inherited Roles
+ - data-hub-common
+ - rest-writer
+
+#### Privileges
+
+Privilege | Notes
+---------|----------
+ ps-user | Needed to record provenance data
+ xdmp-document-load | Needed to load data directly in the database
+ any-uri | Required to write documents to any URI without explicit privileges granted
+ any-collection | Required to write documents to any collection without explicit privileges granted
+
+### hub-central-user
+
+All Data Hub roles intended for Hub Central users inherit this role.
+
+#### Associate Authorities
+ - canLoginToHubCentral
+
+#### Inherited Roles
+ - data-hub-common
+
+### hub-central-downloader
+
+Role to determine if a user can download the Hub Central project.
+
+#### Associate Authorities
+ - canDownloadProject
+
+#### Inherited Roles
+ - data-hub-common
+
+### hub-central-clear-data
+
+Role to determine if a user can clear data through Hub Central.
+
+#### Associate Authorities
+ - canClearData
+
+#### Inherited Roles
+ - data-hub-common-writer
+
+### hub-central-step-runner
+
+Can run steps through Hub Central
+
+#### Associate Authorities
+ - canReadStep
+ - canRunStep
+
+#### Inherited Roles
+ - hub-central-user
+ - data-hub-common-writer
+ - data-hub-flow-reader
+ - data-hub-load-data-reader
+ - data-hub-mapping-reader
+ - data-hub-match-merge-reader
+ - data-hub-step-definition-reader
```

Comment:
IIRC, a hub-central-step-runner should only be able to see the steps in a flow(Run tile) but not the step configurations(Entities tile). So should not inherit load-data/mapping/match-merge/step-definition reader.

Reply:
That is a difference between the `hub-central-`prefixed roles and the `data-hub-` prefixed roles. The `hub-central` roles dictate what can be done through the UI/middle-tier and the `data-hub` roles are primarily document-level permissions. In order for a `hub-central-step-runner` to be able to run a step they need the document-level read permissions to the various artifacts.